### ISSUE TITLE
Collision detection

### ISSUE THREAD

**Issue by fabiantheblind** [basiljs/basil.js#53]
As mentioned in #50 by @trych

**Comment by fabiantheblind** [comment]
Some useful links for collision detection  

2D collision detection - Game development | MDN
https://developer.mozilla.org/en-US/docs/Games/Techniques/2D_collision_detection


Javascript: Collision detection - Stack Overflow
http://stackoverflow.com/questions/2440377/javascript-collision-detection


GitHub · Where software is built
https://github.com/jriecken/sat-js/blob/master/README.md


HTML5 Canvas Game: 2D Collision Detection - Programming Tutorials, Tips, and Tricks
http://blog.sklambert.com/html5-canvas-game-2d-collision-detection/

**Comment by b-g** [comment]
If we had a working `node require` mechanism ... it would be very easy to wrap something like 

```
turf.intersect(turf.polygon(points1), turf.polygon(points2))
or
turf.inside(turf.point(p), turf.polygon(points))
or
turf.inside(turf.point(p), turf.bboxPolygon(bbox))
or
http://turfjs.org/docs/#within
```
inside `b.collides(object, object)`. I think we should avoid to come up with our own implementation if we have a choice.

If a plain bbox intersection check were enough, then https://github.com/mapbox/cheap-ruler#performance might be the best bet.

(NB: check whether turf.js can deal with points outside of the geo realm e.g. longitude +180°)

**Comment by fabiantheblind** [comment]
Yes. Using Node would be great. I've been reading a bit about Adobes HTML panels. They can use node modules in the frontend and then just communicate with the IDAPI via plain old Extendscript. Maybe this is something for a Basiljs v3 :-)  

What we can do is using extendscriptr to build Basil. But we are still at the beginning with extendscriptr and would not recommend using it for production now but we can use npm to at least get the lib ad keep it up to date. Then we would "just" need a build process that:  

- grabs the lib
- polyfills Basil
- concats/bundles all of it.

**Comment by trych** [comment]
Two references for collision detection in JS:

http://www.jeffreythompson.org/collision-detection/
https://github.com/bmoren/p5.collide2D
